flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/CliFrontend.java

```diff
@@ -132,103 +106,6 @@ static CliExecutor createExecutor(CommandLine commandLine) throws Exception {
-            } else {
-                restoreMode = SavepointConfigOptions.RESTORE_MODE.defaultValue();
-            }
-            // allowNonRestoredState is always false because all operators are predefined.
-
-            return (SavepointRestoreSettings)
-                    Arrays.stream(SavepointRestoreSettings.class.getMethods())
-                            .filter(
-                                    method ->
-                                            method.getName().equals("forPath")
-                                                    && method.getParameterCount() == 3)
-                            .findFirst()
-                            .map(
-                                    method -> {
-                                        try {
-                                            return method.invoke(
-                                                    null,
-                                                    savepointPath,
-                                                    allowNonRestoredState,
-                                                    restoreMode);
-                                        } catch (IllegalAccessException
-                                                | InvocationTargetException e) {
-                                            throw new RuntimeException(
-                                                    "Failed to invoke SavepointRestoreSettings#forPath nethod.",
-                                                    e);
-                                        }
-                                    })
-                            .orElseThrow(
-                                    () ->
-                                            new RuntimeException(
-                                                    "Failed to resolve SavepointRestoreSettings#forPath method."));
-        } else {
-            return SavepointRestoreSettings.none();
-        }
-    }
-
-    private static Path getFlinkHome(CommandLine commandLine) {
-        // Check command line arguments first
-        String flinkHomeFromArgs = commandLine.getOptionValue(CliFrontendOptions.FLINK_HOME);
-        if (flinkHomeFromArgs != null) {
-            LOG.debug("Flink home is loaded by command-line argument: {}", flinkHomeFromArgs);
-            return new Path(flinkHomeFromArgs);
-        }
-
-        // Fallback to environment variable
-        String flinkHomeFromEnvVar = System.getenv(FLINK_HOME_ENV_VAR);
-        if (flinkHomeFromEnvVar != null) {
-            LOG.debug("Flink home is loaded by environment variable: {}", flinkHomeFromEnvVar);
-            return new Path(flinkHomeFromEnvVar);
-        }
-
-        throw new IllegalArgumentException(
-                "Cannot find Flink home from either command line arguments \"--flink-home\" "
-                        + "or the environment variable \"FLINK_HOME\". "
-                        + "Please make sure Flink home is properly set. ");
-    }
-
```

Comment:
Move the methods for loading and merging flink config to the FlinkEnvironmentUtils class